gustaf/spline/base.py

```diff
@@ -547,6 +553,100 @@ def nurbs(self):
         return self.copy()
 
 
+class Multipatch(GustafBase, splinepy.Multipatch):
+    __show_option__ = visualize.MultipatchShowOption
+
+    def __init__(
+        self,
+        splines=None,
+        interfaces=None,
+        as_boundary=False,
+    ):
+        """
+        Multipatch
+
+        Parameters
+        ----------
+        splines : list-like
+          List of splines to store as multipatch
+        interfaces : array-like
+          Defines the connectivity in between patches
+        as_boundary : bool
+          Multipatch is a boundary object of a higher dimensional geometry. If
+          set to true, additional checks are performed on the interfaces,
+          requiring strict inter connectivity between all patches
+
+        Returns
+        -------
+        None
+        """
+        splinepy.Multipatch.__init__(
+            self,
+            splines=splines,
+            interfaces=interfaces,
+        )
+        GustafBase.__init__(self)
+        self._show_options = self.__show_option__(self)
+
+    @property
+    def show_options(self):
+        """
+        Show option manager for multipatch systems.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        show_options: MultipatchShowOption
+        """
+        return self._show_options
+
+    def show(self, **kwargs):
+        """ """
+        # Retrieve show options out of valid options
+        spline_list = self.splines.copy()
+        if kwargs.get("boundary_ids", False):
+            bsp = self.boundary_patches
+            bsp_id = np.abs(self.interfaces[self.interfaces < 0])
+
+            # Create a custom color-map
+            max_id = np.max(bsp_id)
+            colors_per_dim = int(np.ceil((1 + max_id) ** (1 / 3)))
+            rgb_color_list = np.reshape(
+                np.meshgrid(
+                    *(np.linspace(0, 1, colors_per_dim) for _ in range(3))
+                ),
+                (3, -1),
+            ).T
+            color_list = [
+                (i + 0.5, rgb_color_list[i, :]) for i in range(max_id + 1)
+            ]
+            lut = build_lut(color_list)
+
+            for patch, id in zip(bsp, bsp_id):
+                patch.splinedata["bid" + str(id)] = BSpline(
+                    degrees=[1] * patch.para_dim,
+                    knot_vectors=np.repeat(
+                        patch.parametric_bounds.T, 2
+                    ).reshape(patch.para_dim, -1),
+                    control_points=[[id]] * (2**patch.para_dim),
+                )
+                patch.show_options["dataname"] = "bid" + str(id)
```

Comment:
is this necessary? can't you just assign a color?

Reply:
I have reworked this part a bit; please look at it. It should be much simpler now without using cmap but has problems with a large number of boundaries. (only visual problems with how different the colors are for different boundaries) We might have to compare the before and after pictures to see if that is a problem or just a problem with humans being unable to distinguish many color variations.